libs/etterlatte-ktor/src/main/kotlin/Ping.kt

```diff
@@ -0,0 +1,61 @@
+package no.nav.etterlatte.libs.ktor
+
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.newSingleThreadContext
+
+interface Pingable {
+    val serviceName: String
+    val beskrivelse: String
+    val endpoint: String
+    suspend fun ping(): PingResult
+
+    @OptIn(DelicateCoroutinesApi::class)
+    fun asyncPing() {
+        GlobalScope.launch(newSingleThreadContext("pingThread")) {
+            ping()
+        }
+    }
+}
+
+class PingResult private constructor(
+    private val serviceName: String,
+    private val status: ServiceStatus,
+    private val endpoint: String,
+    private val beskrivelse: String,
+    private val errorMessage: String?
+) {
+    fun getStatus(): ServiceStatus {
+        return status
+    }
+
+    fun toStringServiceDown(): String {
```

Comment:
Forslag: Ha Up og Down som subklassar av PingResult, og så override toString i Down-klassa.

Reply:
Er det slik at kun Down skal ha tilgang på `toStringServiceDown` metoden? :D

Reply:
Ja, den er jo ikkje relevant for Up, så tenkjer det.

Reply:
Som i grunn er ein fordel, for da tenkjer eg vi kan droppe også errorMessage frå up, sia heller ikkje den bør vera relevant der.

Reply:
Kunne også gjort PingResult sealed hvis du gjør dette.

Har vel lite å si her - men gir oss mulighet til letter "exhaustive" med bruk av when pingResult senere hvis det er et behov